multimodal/src/autogluon/multimodal/data/preprocess_dataframe.py

```diff
@@ -485,6 +491,50 @@ def transform_rois(
 
         return ret_data, ret_type
 
+    def transform_ovd(
+        self,
+        df: pd.DataFrame,
+    ) -> Tuple[Dict[str, List[List[str]]], Dict[str, str]]:
+        """
+        Preprocess image + prompt data.
+        For image data we preprocess them by collecting their paths together. If one sample has multiple images
+        in an image column, assume that their image paths are separated by ";".
+        For rois data we simply convert them from a column of pandas dataframe to a list.
+        This function needs to be called preceding the rois processor in "process_rois.py".
+
+        Parameters
+        ----------
+        df
+            The multimodal pd.DataFrame.
+
+        Returns
+        -------
+        image_features
+            All the image data stored in a dictionary.
+        image_types
+            The column types of these image data, e.g., image_path or image_identifier.
+        """
+        assert (
+            self._fit_called or self._fit_x_called
+        ), "You will need to first call preprocessor.fit_x() before calling preprocessor.transform_ovd."
+
+        x = self.transform_image(df)
+        ret_data = x[0]
+        ret_type = x[1]
+
+        for col_name in self._text_feature_names:
+            col_type = self._column_types[col_name]
+
+            if col_type == TEXT:
+                processed_data = df[col_name].tolist()
+            else:
+                raise ValueError(f"Unknown text type {col_type} for column {col_name}")
+
+            ret_data[col_name] = processed_data
+            ret_type[col_name] = self._column_types[col_name]
```

Comment:
Can we just reuse `transform_text`?

Reply:
It involves some work in text_processor. Will refactor it in next PR.